integration_tests/src/main/python/date_time_test.py

```diff
@@ -130,20 +130,48 @@ def test_datediff(data_gen):
             'datediff(a, date(null))',
             'datediff(a, \'2016-03-02\')'))
 
-@allow_non_gpu(*non_utc_allow)
+@pytest.mark.skipif(not is_supported_time_zone(), reason="not all time zones are supported now, refer to https://github.com/NVIDIA/spark-rapids/issues/6839, please update after all time zones are supported")
 def test_hour():
     assert_gpu_and_cpu_are_equal_collect(
-        lambda spark : unary_op_df(spark, timestamp_gen).selectExpr('hour(a)'))
+        lambda spark : unary_op_df(spark, timestamp_gen).selectExpr('hour(a)'),
```

Comment:
Nit: we can consider merging these two cases together.
```
non_utc_allow = ['ProjectExec'] if is_supported_time_zone() else []
```

Reply:
The one downside of this change is that we don't truly verify that we fell back to the CPU for the operator we care about. Getting the right answer is probably good enough though.